extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/filter/BloomDimFilter.java

```diff
@@ -165,6 +169,23 @@ public boolean applyNull()
               }
             };
           }
+
+          @Override
+          public Predicate<Object[]> makeArrayPredicate(@Nullable TypeSignature<ValueType> arrayType)
+          {
+            final ExpressionType expressionType = arrayType == null || !arrayType.isArray()
```

Comment:
I would hope that in `makeArrayPredicate` we were already past needing to validate that the type passed in is an `arrayType.isArray()`...  It would be nice if there was some class specialization or something that could be done on the actual method signature which would help us in just assuming that it's an array type...

That or the code could just assume it?

Reply:
oops, this check shouldn't be necessary i think because it should only ever be called on things that are definitely arrays, otherwise it would be null. will fix.